merge geomar dev into latest release

seb-wahl opened the pull request:
This merge request has the following changes. I started by merging v6.54.5 into geomar_dev:
* adjusted all the yaml files after the removal of the environment_changes stuff. 
* allow more flexibility in echam when it comes to restarts: the output streams and restart streams can now differ. I defined the default output streams to match the restart streams. Removes a lot of confusing warnings for our setups
* fixed a bug with a `dr_hook_...*` var not found anymore after the removal of the environment_changes stuff. 
* remove deprecated env var declarations that caused a lot of warnings
* updated slurm.yaml to allow for the usage of mpirun (which we currently need on glogin), please check but I can't see any problems merging this
* included our modifications to `oasis.py`. This needs to be reviewed carefully. Please see my comments in `oasis.py`.
* includes foci-mops_lmu, a setup that includes an updated version of JSBACH 3.20p1 used and LMU Munich and now in Kiel within the RESCUE project. It allows to simulate carbon dioxide removal methods in JSBACH using a 12th tile. This means I had to make the static 11 tiles in jsbach.yaml a variable `${tiles}`

Known bugs:
* a `-finished_config.yaml` remains in the directory where I run esm_master and I get the following "error" message:
```bash
Build command finished on Wed May 14 08:37:43 2025.
/mnt/vast-nhr/home/sebastian.wahl01/u11179/esm/models/test/foci-mops_lmu/nemo-ORCA05_LIM2_FOCI_MOPS_OASISMCT4/CONFIG
/mnt/vast-nhr/home/sebastian.wahl01/u11179/esm/models/test/foci-mops_lmu
Problems moving ``./foci-mops_lmu-finished_config.yaml`` to ``./foci-mops_lmu/finished_config.yaml``
(esm-tools) [shk00060] u11179@glogin6 test $ ls
-finished_config.yaml  foci-mops_lmu
```

seb-wahl committed:
first draft of FOCI-MOPS with ECHAM6 from LMU Munich
compiles and starts, but misses (as expected) information in existing
JSBACH restart files. Cold start not yet implemented

seb-wahl committed:
added a first draft of runscript for LMU edition
links all currently known files correctly
running not yet tested as info from Shradda is still missing

seb-wahl committed:
made number of tiles a variable

seb-wahl committed:
added missing info in xios.yaml
draft version of ECHAM76-LMU that start up correctly
but still fails due to land sea mask mismatch

seb-wahl committed:
added namelists templates (or links)

seb-wahl committed:
added missing ;

seb-wahl committed:
fixed dynveg file placeholder

seb-wahl committed:
fix echam lmu compilation when submodule has already been cloned

seb-wahl committed:
latest version of LMU runscript, still crashes

seb-wahl committed:
FOCI-MOPS LMU runs including updated output for jsbach and echam
fixed restart setup in a hacky way (see foci.yaml)
added a new runscript for FOCIOIFS

seb-wahl committed:
Merge remote-tracking branch 'origin/feature/add-foci-mops-lmu' into geomar_dev
exludes the conflicting oasis.py

seb-wahl committed:
manually add the modifications in oasis.py again
This should be the only difference between geomar_dev and release that matters

seb-wahl committed:
fixed merge error in foci.yaml

seb-wahl committed:
renamed nest restart labels to fix Sayantanis restart issues

seb-wahl committed:
added Joakims additions to slurm.py which are required to use mpirun on glogin

seb-wahl committed:
added hotfix for oasis_time_stamp support needed by focioifs

seb-wahl committed:
added postprocessing for RESCUE (temporary, should go somewhere else)
allow different streams for restart and output for ECHAM and JSBACH

seb-wahl committed:
fix paths in echam_monitoring

seb-wahl committed:
fixed all bugs that occurred before the first longer run
with FOCI-MOPS with the 12 tile edition

seb-wahl committed:
added working FOCILMU piCtl setup

seb-wahl committed:
updated pp echam RESCUE, now netCDF

seb-wahl committed:
increased compression to zip_3

seb-wahl committed:
Merge pull request #1350 from esm-tools/geomar_dev

merge geomar dev into geomar branch

seb-wahl committed:
separate echam6 namelist templates for foci and foci_lmu

seb-wahl committed:
updated settings for NEMO4 standalone done a loooong time ago

seb-wahl committed:
Merge pull request #1351 from esm-tools/geomar_dev

separate echam6 namlist templates for foci and foci_lmu

seb-wahl committed:
manually resolve conflicts between release and geomar_dev
before merging this branch into geomar_dev
differences where too large for an efficient direct merge

seb-wahl committed:
manual merge of selected yaml from release to avoid conflicts

seb-wahl committed:
Merge branch 'release_geomar_dev_intermediate' into geomar_dev

seb-wahl committed:
fixed indentation bug in jsbach.yaml, remove deprecated env var declaration

seb-wahl committed:
re-added missing foci_lmu block

seb-wahl committed:
fix dr_hook var not found in oifs
updated env var settings to get rid of the deprecation warning

seb-wahl commented:
just found another bug, so don't start the merge yet

seb-wahl committed:
added missing xios version info for foci again, bugfix in nemo postprocessing

seb-wahl commented:
Bugs are fixed. Feel free to test and merge. Let me know if you need help

seb-wahl committed:
Merge branch 'release' into geomar_dev

seb-wahl committed:
updated cmake intel compiler to new minor version as 2023.2.1 disappeared
still not working as a matching intelmpi is still missing, waiting for support to reply

seb-wahl committed:
fixed a choose inconsistency detected by updated ESM-Tools
added OIFS vars to runoffmapper for focioifs compilation
updated foci-mops-lmu-scenario-1d.yaml
fixed tiny bug in nemo_postprocessing.sh

seb-wahl committed:
Merge remote-tracking branch 'origin/release' into geomar_dev

seb-wahl committed:
reverted change in intel compiler version for use with lmod-nhr

seb-wahl committed:
minor updates to two sample runscripts

seb-wahl committed:
Merge branch 'release' into geomar_dev

seb-wahl committed:
Merge branch 'geomar_dev' into geomar

seb-wahl committed:
set a default version for xios again
add OIFS variables again as the rnfma version used in FOCIOIFS still requires those

seb-wahl committed:
updated path to new $COLD storage on glogin in all runscripts that have
the path hardcoded, added FOCIOIFS-AGRIF example from Sayantani Ohja

seb-wahl committed:
Merge branch 'geomar_dev' into geomar

seb-wahl committed:
WIP: added modifications from Rene to latest geomar_dev
branch locally, still requires testing on levante

seb-wahl committed:
WIP: integrate ICON-FESOM into ESM-Tools

seb-wahl committed:
fix bugs that were introduced during the manual merge of focioifs-5.0
into the latest geomar_dev

seb-wahl committed:
hotfix: remove version tag from xios and add it to some of the focioifs runscripts

seb-wahl committed:
Merge branch 'geomar_dev' into feature/focioifs-5.0

seb-wahl committed:
Merge branch 'geomar_dev' into feature/icon-fesom

seb-wahl committed:
WIP: still not running, but most files for ICON are now found, nothing done for FESOM yet

seb-wahl committed:
latest version of FOCIOIFS-5.0
compiles and starts, segfault in xios ini
gcc + openmpi

seb-wahl committed:
fixed oifs43-v2 setup on glogin and added example
updated RESCUE pi-ctl template
updated foci-mops-ESM example runscript

seb-wahl committed:
Merge branch 'geomar_dev' into geomar

seb-wahl committed:
add support for post_run_commands to solve (or call it a workaround) for
https://github.com/esm-tools/esm_tools/issues/1148

seb-wahl committed:
added example for post_run_commands

seb-wahl committed:
Merge branch 'geomar_dev' into geomar

seb-wahl committed:
fix bug in echam_postprocessing_RESCUE.sh

seb-wahl committed:
Merge branch 'geomar_dev' into geomar

seb-wahl committed:
latest setup, still hangs upon coupling initialisation

seb-wahl committed:
added clean support for intel again for focioifs components
removed OIFS* env vars from runoffmapper

seb-wahl committed:
WIP: compiles, runscript still shows a lot of missing files

seb-wahl committed:
updated example for oifs-43r3-tco95 with 1m output on glogin

seb-wahl committed:
Merge branch 'geomar_dev' into geomar

seb-wahl committed:
added nc-time-axis and adjusted monitoring accordingly

seb-wahl committed:
latest updates for Sabine and Matthias

seb-wahl committed:
MOZ runs with 1h coupling and fixed postprocessing

seb-wahl committed:
Merge remote-tracking branch 'origin/geomar_dev' into feature/echam6-moz

seb-wahl committed:
added missing land file tag for moz

seb-wahl committed:
add land tag to land_file_tags

seb-wahl committed:
undo previous commit

seb-wahl committed:
merge release into geomar_dev

seb-wahl committed:
Merge branch 'geomar_dev' into geomar
preliminary support for focioifs-5.0 (WIP)
added support for foci-moz (echemistry in ECHAM6)

seb-wahl committed:
updated from geomar_dev (v6.58)

seb-wahl committed:
ICON-FESOM technically running
no restart yet as ICON-FESOM finishes with an exit code
no output handling yet

seb-wahl committed:
IMPORTANT: CONSERV coupling vs. GAUSWGT in foci-moz and foc-default_mct5
see runscripts/foci/README_important.txt for details
add temraw output for SOLVe project in foci-moz

seb-wahl committed:
turn error into warning for icon-fesom for now

seb-wahl committed:
updated default settings, namelist.echam for moz updated

seb-wahl committed:
hopefully fixed the ICON restart issue

seb-wahl committed:
WIP: updated restart settings

seb-wahl committed:
WIP: updated restart settings 2

seb-wahl committed:
fixed whitespace in icon.yaml

seb-wahl committed:
bugfixes, cleanup
compiling, running and restart now work

seb-wahl committed:
change timestep to 900s, switch to Intel

seb-wahl committed:
Merge remote-tracking branch 'origin/release' into feature/icon-fesom

seb-wahl committed:
added file movements for FESOM, added documentation

seb-wahl committed:
fixed typo

seb-wahl committed:
updated ICON-FESOM instructions

seb-wahl committed:
added compilation support for moz with MCT2.8
added missing coupling dir for foci-moz

seb-wahl committed:
fixed typos for moz mct2 autotools compile

seb-wahl committed:
revert compile settings for echam with mct4 or 5

seb-wahl committed:
Merge branch 'hotfix/echam_compile' into geomar_dev

Commit:
update icon-fesom-v2

seb-wahl committed:
added support for intel 2021 classic without using the deprecated nhr-lmod setup on glogin

seb-wahl committed:
added preliminary support for intel2025, echam compiles (with changes to CMakeLists.txt)
XIOS still failing
fixed RESCUE postprocessing by adding the mandatory new option -l and -a to
echam_postprocessing_RESCUE.sh that came in when MOZ was added

mandresm commented on configs/components/icon/icon.yaml:
```diff
@@ -1,381 +1,536 @@
+            active: ${lcarbon}
+
+   choose_jsbach_with_hd:
+      yes:
```
Would it make sense to change this to `true/false`? Just to remain consistent to all of our other switches that normally use booleans.

mandresm commented on configs/components/icon/icon.yaml:
```diff
@@ -1,381 +1,536 @@
+               iolibraries: None 
+               compiler_mpi: None
+
+               choose_icon.version:
+                  2.6.6-nwp:
+                     input_dir: /pool/data/ICON
+                     icon_pool: ${input_dir}/grids/private/mpim/icon_preprocessing/source
+                     grid_dir: ${input_dir}/grids
+                     aerosol_dir: ${icon_pool}/aerosol_29-06-2018/kinne/${resolution}_${icon.grid_label}
+                     jsbach_ic_dir: ${input_dir}/grids/private/jsbach/mpim/${grid_id}/land/r0002
+                     jsbach_forcing_dir: ${jsbach_ic_dir}
```
I would move most of this block into the `icon` section itself, instead of keeping it  inside `computer`. The only parameter that seems computer-dependent is `input_dir` and that could probably be made independent by making it be:

```yaml
    input_dir: ${computer.pool_dir}/ICON
```

mandresm commented on configs/components/oifs/oifs.yaml:
```diff
@@ -348,12 +362,12 @@ oifs:
         destination: oifs-48r1
         with_xios: false
         major_version: 48r1
+
```
```suggestion
```

mandresm commented on configs/components/oifs/oifs.yaml:
```diff
@@ -1300,6 +1316,10 @@ oifs:
             ifsdata: ifsdata/*
             tl_data: ${res_number_tl}/*
             o3_data: ${res_number_tl}/o3chem_l${nlev}/*
+            uwavein: uwavein
+            specwavein: specwavein
+            sfcwindin: sfcwindin
+            cdwavein: cdwavein
```
```suggestion
```

These files shouldn't be needed here as they are added when `wam` is turned on:

https://github.com/esm-tools/esm_tools/blob/51ef3711e3abb0721417d86556273e7167b8d08f/configs/components/oifs/oifs.yaml#L915-L927

mandresm commented on configs/components/oifs/oifs43.env.yaml:
```diff
@@ -351,46 +351,86 @@ computer:
-            OIFS_CFLAGS: '"-fp-model precise -O3 -g -traceback -qopt-report=0 -fpe0 -qopenmp -march=core-avx2 -mtune=core-avx2"' # -qoverride-limits -fast-transcendentals -m64 -fma -pc64"'
-            OIFS_CCDEFS: '"LINUX LITTLE INTEGER_IS_INT _ABI64 BLAS _OPENMP"'
+          choose_compiler_mpi: 
+            intel2022_openmpi:
```
check that this does not break anything for AWICM3 (in the esm_tests in GitHub)

mandresm commented on configs/components/rnfmap/rnfmap.env.yaml:
```diff
@@ -9,6 +9,19 @@
 computer:
   choose_general.execution_mode:
     compile:
+       #add_export_vars:
+        # oasis
+        #OIFS_OASIS_BASE: '$(pwd)/oasis'
+        #OIFS_OASIS_INCLUDE: '"-I$OIFS_OASIS_BASE/build/lib/psmile -I$OIFS_OASIS_BASE/build/lib/psmile/scrip -I$OIFS_OASIS_BASE/build/lib/psmile/mct -I$OIFS_OASIS_BASE/build/lib/psmile/mct/mpeu"'
+        #OIFS_OASIS_LIB: '"-L$OIFS_OASIS_BASE/build/lib/psmile -L$OIFS_OASIS_BASE/build/lib/psmile/scrip -L$OIFS_OASIS_BASE/build/lib/psmile/mct -L$OIFS_OASIS_BASE/build/lib/psmile/mct/mpeu -lpsmile -lmct -lmpeu -lscrip"'
+      # NetCDF
+      #OIFS_NETCDF_INCLUDE: '-I$NETCDFROOT/include'
+      #OIFS_NETCDF_LIB: '"-L$NETCDF_DIR/lib -lnetcdf"'
+      #OIFS_NETCDFF_INCLUDE: '-I$NETCDFFROOT/include'
+      #OIFS_NETCDFF_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
+      # compiler and compiler flags
+      #OIFS_FC: $FC
+
```
```suggestion
```

mandresm commented on configs/machines/levante.yaml:
```diff
@@ -72,7 +72,8 @@ computer:
     # kh 15.02.22 preferred on levante (at the moment), on mistral intelmpi was the default
-    compiler_mpi: intel2022_openmpi
+    #compiler_mpi: intel2022_openmpi
+    compiler_mpi: gnu11_ompi4 
```
```suggestion
    compiler_mpi: intel2022_openmpi 
```

This might break AWI models because they might all depend on this default. Either, make sure Geomar models setup this in their relevant files and revert this change or check that AWI models all specify their `compiler_mpi` parameter in their files. Check esm_tests changes

mandresm commented on configs/machines/levante.yaml:
```diff
@@ -240,6 +243,7 @@ computer:
                 - "load netcdf-fortran/4.5.3-openmpi-4.1.2-gcc-11.2.0"
                 - "load git/2.31.1-gcc-11.2.0"
                 - "load libaec/1.0.5-gcc-11.2.0"
+
```
```suggestion
```

mandresm commented on configs/machines/levante.yaml:
```diff
@@ -177,6 +180,7 @@ computer:
                             MPI_LIB: "\"$(mpiifort -show |sed -e 's/^[^ ]*//' -e 's/-[I][^ ]*//g')\""
 
            intel2022_openmpi:
+                    python_version: "python3/2023.01-gcc-11.2.0"
```
I think I remember this possibly breaking things for AWICM3. I'll need to check.

mandresm commented on configs/machines/levante.yaml:
```diff
@@ -264,7 +269,7 @@ computer:
                 LD_LIBRARY_PATH: $MPI_HOME/lib:$NetCDF_C_ROOT/lib:$NetCDF_Fortran_ROOT/lib:$HDF5ROOT/lib:$AEC_ROOT/lib:$TBBMALLOC_DIR:$LD_LIBRARY_PATH
                 LD_LIBRARY_PATH[(0)]: /sw/spack-levante/intel-oneapi-mkl-2022.0.1-ttdktf/mkl/2022.0.1/lib/intel64:$LD_LIBRARY_PATH
-                # For OASIS3-MCT5 from CERFACS
+                # For OASIS3-MCT5 from CERFACS      
```
```suggestion
                # For OASIS3-MCT5 from CERFACS
```

mandresm commented on configs/machines/levante.yaml:
```diff
@@ -308,29 +331,14 @@ computer:
-
-                # The following line is also moved to echam.yaml, only active for mistral and paleodyn echam6:
-                # - "LD_LIBRARY_PATH=/sw/rhel6-x64/netcdf/parallel_netcdf-1.6.1-impi-intel14/lib/:$LD_LIBRARY_PATH"
+                LD_LIBRARY_PATH[(0)]: $HDF5ROOT/lib:$NETCDFROOT/lib:$NETCDFFROOT/lib:$AEC_ROOT/lib:$LD_LIBRARY_PATH
```
I don't think this would break anything for AWI, but let's check. Otherwise, we will need to move the changes required by icon into ICON itself. To be discussed with @seb-wahl

mandresm commented on src/esm_parser/dict_to_yaml.py:
```diff
@@ -160,11 +160,15 @@ def add_eol_comments_with_provenance(commented_config, config):
-                    provenance_comment = f"{provenance['yaml_file']},line:{provenance['line']},col:{provenance['col']}"
-                else:
-                    provenance_comment = f"no provenance info"
+                # due to duplicate namelists (occur with ICON) we get an 
+                # IndexError: list index out of range
+                # workaround by seb-wahl
+                # TODO: find a better solution
+                provenance_comment = f"no provenance info"
+                if 0 <= indx < len(config): 
+                    provenance = getattr(config[indx], "provenance", [None])[-1]
+                    if provenance:
+                        provenance_comment = f"{provenance['yaml_file']},line:{provenance['line']},col:{provenance['col']}"
```
I need to debug this, there might be a way of getting rid of this problem. It's strange that this does not happen for ECHAM

mandresm commented on src/esm_runscripts/batch_system.py:
```diff
@@ -352,6 +352,42 @@ def get_bash_command_to_print_in_progress_log(config, subjob, message):
+    def get_post_run_commands(config):
+        extras = []
+
+        # Search for ``post_run_commands``s in the components
+        for component in config.keys():
+            post_run_commands = config[component].get("post_run_commands")
+            if isinstance(post_run_commands, list):
+                for pr_command in post_run_commands:
+                    if isinstance(pr_command, str):
+                        extras.append(pr_command)
+                    else:
+                        user_error(
+                            'Invalid type for "post_run_commands"',
+                            (
+                                f'"{type(pr_command)}" type is not supported for '
+                                + f'elements of the "post_run_commands", defined in '
+                                + f'"{component}". Please, define '
+                                + '"post_run_commands" as a "list" of "strings" or a "list".'
+                            ),
+                        )
+            elif isinstance(post_run_commands, str):
+                extras.append(post_run_commands)
+            elif post_run_commands == None:
+                continue
+            else:
+                user_error(
+                    'Invalid type for "post_run_commands"',
+                    (
+                        f'"{type(post_run_commands)}" type is not supported for '
+                        + f'"post_run_commands" defined in "{component}". Please, define '
+                        + '"post_run_commands" as a "string" or a "list" of "strings".'
+                    ),
+                )
+        return extras
```
```suggestion
        # Search for ``post_run_commands``s in the components
        for component in config.keys():
            post_run_commands = config[component].get("post_run_commands")
            if isinstance(post_run_commands, str):
                post_run_commands = [post_run_commands]
            elif isinstance(post_run_commands, list):
                pass
            elif post_run_commands == None:
                continue
            else:
                user_error(
                    'Invalid type for "post_run_commands"',
                    (
                        f'"{type(post_run_commands)}" type is not supported for '
                        + f'"post_run_commands" defined in "{component}". Please, define '
                        + '"post_run_commands" as a "string" or a "list" of "strings".'
                    ),
                )
            for pr_command in post_run_commands:
                if isinstance(pr_command, str):
                    extras.append(pr_command)
                else:
                    user_error(
                        'Invalid type for "post_run_commands"',
                        (
                            f'"{type(pr_command)}" type is not supported for '
                            + f'elements of the "post_run_commands", defined in '
                            + f'"{component}". Please, define '
                            + '"post_run_commands" as a "list" of "strings" or a "list".'
                        ),
                    )
        return extras
```

mandresm replied:
TODO for @mandresm: make it the same for `pre_run_commands` and generalize it in a single function that can be called both for pre and post run.

mandresm commented on src/esm_runscripts/batch_system.py:
```diff
@@ -554,6 +590,11 @@ def write_simple_runscript(config, cluster, batch_or_shell="batch"):
                 ]["next_submit"]
 
+                # extra entries for each subjob
+                post_run_commands = batch_system.get_post_run_commands(config)
```
```suggestion
                post_run_commands = batch_system.get_commands(config, "post_run_commands")
```
TODO for @mandresm: change it to something like this.

mandresm commented on src/esm_runscripts/coupler.py:
```diff
@@ -11,23 +11,23 @@ class coupler_class:
-                    list(full_config[self.name]["coupling_input_fields"])
-                )
         if name == "oasis3mct":
+            self.process_ordering = full_config[name]["process_ordering"]
+            self.coupled_execs = []
+            for exe in self.process_ordering:
+                self.coupled_execs.append(full_config[exe]["executable"])
+            self.runtime = full_config["general"]["runtime"][5]
+            self.nb_of_couplings = 0
+            if "coupling_target_fields" in full_config[self.name]:
+                for restart_file in list(full_config[self.name]["coupling_target_fields"]):
+                    self.nb_of_couplings += len(
+                        list(full_config[self.name]["coupling_target_fields"][restart_file])
+                    )
+            if "coupling_input_fields" in full_config[self.name]:
+                for restart_file in list(full_config[self.name]["coupling_input_fields"]):
+                    self.nb_of_couplings += len(
+                        list(full_config[self.name]["coupling_input_fields"])
+                    )
```
TODO for @mandresm: if this belongs exclusively to oasis, move it to the `oasis.py`

mandresm commented on src/esm_runscripts/coupler.py:
```diff
@@ -42,30 +42,22 @@ def __init__(self, full_config, name):
-        if coupler_name == "yac":
-            couplingfile = "coupling.xml"
-        else:
+        couplingfile = None
```
TODO @mandresm: check for attributes in self. If none, then just return (make it oasis independend)

seb-wahl committed:
set default compiler for icon-fesom correctly

seb-wahl committed:
set a reasonable value for pool_dir

mandresm reviewed (changes requested):
@seb-wahl, as mentioned in our in-person meeting, I finished this review even if this is not meant to be merged in `release` so that we can use some of this comments as a basis for the PR og `geomar` branch into `release`.

seb-wahl committed:
copying instead of linking namelists

seb-wahl committed:
fixed bug in atm and land file tags

seb-wahl committed:
Merge remote-tracking branch 'origin/geomar_dev' into feature/echam6-moz

seb-wahl committed:
fix duplicate entry atm_file_tags

seb-wahl committed:
modifications to reproduce Franzi Schwarzkopf's stage

seb-wahl committed:
removed surf from moz land_file_tags

seb-wahl committed:
Merge branch 'geomar_dev' into feature/focioifs-5.0

seb-wahl committed:
remove surf instead of yasso (see prev commit)

seb-wahl committed:
Merge branch 'feature/echam6-moz' into feature/focioifs-5.0

seb-wahl committed:
added foci-moz example

seb-wahl committed:
limit runtime to two years in example script

seb-wahl committed:
Merge branch 'feature/echam6-moz' into feature/icon-fesom

seb-wahl committed:
Updated settings for RESCUE
!!! nproma = 32 in runscripts/foci/RESCUE/foci-mops-lmu-scenario.yaml !!!

seb-wahl committed:
fixed path issue with monitoring
fixed date issue for date error date -d 19160501 +1month
on levante by using date -u

seb-wahl committed:
Merge branch 'feature/echam6-moz' into feature/icon-fesom

seb-wahl committed:
added piCtl setup for ECHAM6-MOZ including example runscript

seb-wahl committed:
Merge branch 'feature/echam6-moz' into feature/icon-fesom

seb-wahl committed:
used bc_ as prefix everywhere to be consistent
update icon settings to have Sveta's exp0009 settings
still crashes (but with the same as Nils' crash)

seb-wahl committed:
added missing description for levante pp, that may cause problems if pp is actually used

seb-wahl committed:
Merge remote-tracking branch 'origin/feature/echam6-moz' into feature/rescue

seb-wahl committed:
Merge remote-tracking branch 'origin/feature/echam6-moz' into feature/rescue

seb-wahl committed:
separate namelist template for oifs48r1-foci5

seb-wahl committed:
restore fort.4 in namelists/oifs/48r1 to avoid trouble for AWI (see previous commit)

seb-wahl committed:
Merge branch 'feature/focioifs-5.0' into geomar_dev

seb-wahl committed:
fixed ice restart bug nemo5
fixed oifs restarts
fixed handling of context_ifs.xml
fixed fort.4 settings
updated example runscript

seb-wahl committed:
updated runscripts for correct handling of OIFS output

seb-wahl committed:
add focioifs-agrif-5.0
compiles, no runscript yet

seb-wahl committed:
WIP: include OIFS48r1 with NEMO5 and AGRIF (eINALT20) support

seb-wahl committed:
updated runscript for Franzi

Commit:
remove M01 atm->agr fields from coupling. And switch to rnfmap-agrif for focioifs-agrif-5.0

Commit:
cleaning up remapping methods for focioifs. Also move a lot remap definitions to choose statements for easy switching

Commit:
remove opac_fields for 5.0-agrif and use only opat_fields. Also define JK26 runoff

Commit:
fixing coupling for focioifs-agrif-5

seb-wahl committed:
fixed indentation for levante
made arch file choice more flexible

seb-wahl committed:
added updates xios.yaml from feature/focioifs-5.0

seb-wahl committed:
Merge branch 'feature/rescue' into feature/focioifs-5.0

seb-wahl committed:
add git config core.fileMode false for oifs48r1-foci5

seb-wahl committed:
added daily restart template for tests, workaround with pre_run_command for file naming

seb-wahl committed:
updated settings for system_intel_libs on levante

seb-wahl committed:
added runscripts for hist and perpetual year (the latter to be tested)

seb-wahl committed:
Merge remote-tracking branch 'origin/feature/echam6-moz' into geomar_dev

seb-wahl committed:
Merge branch 'feature/focioifs-5.0' into geomar_dev

seb-wahl committed:
Merge remote-tracking branch 'origin/release' into geomar_dev

seb-wahl committed:
generalized all_machines.yaml for Emmy p2 and p3
example runscript (foci-default) on how to use input from lustre-rzg on Emmy p3

seb-wahl committed:
generalize base_dir in example

seb-wahl committed:
Merge branch 'geomar_dev' into feature/rescue

seb-wahl committed:
added namelist templates with tracer
made the new namelists default for an oasismct4_mops based setup

seb-wahl commented:
closed, replaced by https://github.com/esm-tools/esm_tools/pull/1443
